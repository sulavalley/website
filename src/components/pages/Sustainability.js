import React from 'react';

const Sustainability = () => {
  return (
    <div>
      <h1>Estas en Sostenibilidad</h1>
    </div>
  );
};

export default Sustainability;
