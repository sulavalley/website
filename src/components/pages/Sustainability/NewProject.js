import React from 'react';
import base from '../../../base';
import { Button, Form } from 'semantic-ui-react';

class NewProject extends React.Component {
  state = {
    name: '',
    url: '',
    description: ''
  };

  addProject() {
    base.push('projects', {
      data: {
        name: this.state.name,
        url: this.state.url,
        description: this.state.description
      },
      then(err) {
        if (!err) {
          alert('El proyecto a sido aÃ±adido');
        } else {
          alert('Ha ocurrido un error');
        }
      }
    });
  }

  handleSubmit = event => {
    event.preventDefault();
    this.addProject();
  };

  render() {
    const { name } = this.state;
    const { url } = this.state;
    const { description } = this.state;

    return (
      <Form>
        <br />
        <Form.Field>
          <label>Nombre del Proyecto</label>
          <input
            type="text"
            value={name}
            placeholder="Nombre del proyecto"
            onChange={event => this.setState({ name: event.target.value })}
          />
        </Form.Field>
        <Form.Field>
          <label>Imagen</label>
          <input
            type="url"
            value={url}
            placeholder="Enlace de la Imagen"
            onChange={event => this.setState({ url: event.target.value })}
          />
        </Form.Field>
        <Form.Field>
          <label>Descripcion</label>
          <input
            type="text"
            value={description}
            placeholder="Descripcion del Proyecto"
            onChange={event =>
              this.setState({ description: event.target.value })
            }
          />{' '}
        </Form.Field>
        <Button type="submit" id="myButton" onClick={this.handleSubmit}>
          Crear
        </Button>
      </Form>
    );
  }
}

export default NewProject;
