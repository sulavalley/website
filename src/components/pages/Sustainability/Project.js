import React from 'react';
import { Item } from 'semantic-ui-react';

class Project extends React.Component {
  render() {
    const { name, url, description } = this.props;

    return (
      <Item>
        <Item.Image src={url} size="small" />
        <Item.Content>
          <Item.Header>{name}</Item.Header>
          <Item.Description>{description}</Item.Description>
        </Item.Content>
      </Item>
    );
  }
}

export default Project;
