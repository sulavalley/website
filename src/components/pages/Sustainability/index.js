import React, { Component } from 'react';
import {
  Modal,
  Form,
  Input,
  Button,
  TextArea,
  Item,
  Divider,
  Dimmer,
  Loader,
  Segment
} from 'semantic-ui-react';
import Project from './Project';
import base from '../../../base';

class Sustainability extends Component {
  state = {
    projects: [],
    projectModalIsOpen: false,
    newProject: {
      name: '',
      url: '',
      description: ''
    },
    loading: false
  };

  componentDidMount() {
    this.setState({ loading: true });
    base
      .fetch('projects', {
        context: this,
        asArray: true
      })
      .then(data => this.setState({ projects: data }))
      .finally(() => this.setState({ loading: false }))
      .catch(() => console.error('Something went terribly wrong'));
  }

  handleChange = event => {
    const { newProject } = this.state;
    this.setState({
      newProject: { ...newProject, [event.target.name]: event.target.value }
    });
  };

  handleSubmit = () => {
    const { newProject } = this.state;
    base
      .push('projects', {
        data: { ...newProject }
      })
      .then(() => this.setState({ projectModalIsOpen: false }))
      .catch(() => console.error('Something went terribly wrong'));
  };

  render() {
    const { projectModalIsOpen, projects, loading } = this.state;
    const { currentUser } = this.props;
    return (
      <Segment basic>
        <Dimmer active={loading} inverted>
          <Loader size="large">Cargando</Loader>
        </Dimmer>
        {currentUser && (
          <Button
            primary
            onClick={() => this.setState({ projectModalIsOpen: true })}
          >
            Nuevo proyecto
          </Button>
        )}
        <Divider hidden />
        <Item.Group divided>
          {projects.map((project, i) => (
            <Project
              key={i}
              name={project.name}
              url={project.url}
              description={project.description}
            />
          ))}
        </Item.Group>
        <ImageModal
          open={projectModalIsOpen}
          onClose={() => this.setState({ projectModalIsOpen: false })}
          onSubmit={this.handleSubmit}
          onChange={this.handleChange}
        />
      </Segment>
    );
  }
}

const ImageModal = ({ open, onClose, onSubmit, onChange }) => (
  <Modal closeIcon open={open} onClose={onClose}>
    <Modal.Header>Nuevo proyecto</Modal.Header>
    <Modal.Content>
      <Form onSubmit={onSubmit}>
        <Form.Field>
          <label>Nombre</label>
          <Input
            onChange={onChange}
            name="name"
            placeholder="Ingrese un nombre"
          />
        </Form.Field>
        <Form.Field>
          <label>Imagen</label>
          <Input
            onChange={onChange}
            name="url"
            placeholder="Ingrese un enlace a la imagen"
          />
        </Form.Field>
        <Form.Field>
          <label>Descripción</label>
          <TextArea
            onChange={onChange}
            name="description"
            placeholder="Ingrese una descripción"
          />
        </Form.Field>
        <Button
          style={{ marginBottom: 15 }}
          floated="right"
          positive
          type="submit"
        >
          Publicar
        </Button>
      </Form>
    </Modal.Content>
  </Modal>
);

export default Sustainability;
