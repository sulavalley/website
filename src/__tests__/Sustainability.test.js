import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Sustainability from './../components/pages/Sustainability';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Sustainability />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders without crashing', () => {
  shallow(<Sustainability />);
});

test('renders correctly', () => {
  const tree = renderer.create(<Sustainability />).toJSON();
  expect(tree).toMatchSnapshot();
});
