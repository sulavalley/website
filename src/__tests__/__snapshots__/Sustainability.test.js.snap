// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="ui attached tabular menu"
  >
    <a
      className="active item"
      onClick={[Function]}
    >
      Crear Proyecto
    </a>
    <a
      className="item"
      onClick={[Function]}
    >
      Proyectos
    </a>
  </div>
  <form
    action={undefined}
    className="ui form"
    onSubmit={[Function]}
  >
    <br />
    <div
      className="field"
    >
      <label>
        Nombre del Proyecto
      </label>
      <input
        onChange={[Function]}
        placeholder="Nombre del proyecto"
        type="text"
        value=""
      />
    </div>
    <div
      className="field"
    >
      <label>
        Imagen
      </label>
      <input
        onChange={[Function]}
        placeholder="Enlace de la Imagen"
        type="url"
        value=""
      />
    </div>
    <div
      className="field"
    >
      <label>
        Descripcion
      </label>
      <input
        onChange={[Function]}
        placeholder="Descripcion del Proyecto"
        type="text"
        value=""
      />
       
    </div>
    <button
      className="ui button"
      disabled={undefined}
      id="myButton"
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
      type="submit"
    >
      Crear
    </button>
  </form>
</div>
`;
